src/domain/schemas/document-schemas.js

```diff
@@ -71,29 +71,29 @@ export const documentSectionDBSchema = joi.object({
 export const documentRevisionDBSchema = joi.object({
   _id: idOrKeySchema.required(),
   documentId: idOrKeySchema.required(),
-  roomId: idOrKeySchema.allow(null),
+  roomId: idOrKeySchema.allow(null).required(),
   order: joi.number().required(),
   createdOn: joi.date().required(),
   createdBy: idOrKeySchema.required(),
   title: joi.string().required(),
   description: joi.string().allow('').max(maxDocumentDescriptionLength).required(),
-  slug: slugSchema,
+  slug: slugSchema.required(),
   language: joi.string().case('lower').required(),
   sections: joi.array().items(documentSectionDBSchema).required(),
-  restoredFrom: joi.string().allow(null).allow('').required(),
+  restoredFrom: joi.string().allow(null).required(),
   tags: joi.array().items(joi.string()).required(),
-  review: joi.string().allow(null).allow(''),
-  verified: joi.boolean(),
-  allowedOpenContribution: joi.string().valid(...Object.values(DOCUMENT_ALLOWED_OPEN_CONTRIBUTION)),
+  review: joi.string().allow('').required(),
+  verified: joi.boolean().required(),
+  allowedOpenContribution: joi.string().valid(...Object.values(DOCUMENT_ALLOWED_OPEN_CONTRIBUTION)).required(),
   archived: joi.boolean().required(),
   origin: joi.string().required(),
-  originUrl: joi.string().allow(null).allow(''),
+  originUrl: joi.string().allow(null).required(),
```

Comment:
This is an edge case, the URL is not editable by the user, but set by the system to the import origin, so I went with value or `null`.

Reply:
Makes sense. And yes, will be cleaned up soon enough!